cloud/pod_util.go

```diff
@@ -171,17 +173,25 @@ func ExportECSContainerResources(info pod.ContainerResourceInfo) cocoa.ECSContai
 	return *res
 }
 
-// ExportPodContainerDef exports the ECS pod container definition into the equivalent cocoa.ECSContainerDefintion.
-func ExportPodContainerDef(opts pod.TaskContainerCreationOptions) (*cocoa.ECSContainerDefinition, error) {
+// ExportPodContainerDef exports the ECS pod container definition into the
+// equivalent cocoa.ECSContainerDefintion.
+func ExportPodContainerDef(settings *evergreen.Settings, p *pod.Pod) (*cocoa.ECSContainerDefinition, error) {
```

Comment:
A little late to the game here, but why do we pass in the evg settings? Can't we just fetch those from the global env?

Reply:
This is the standard way to use the Evergreen settings (especially in Amboy jobs), because calling `evergreen.GetEnvironment()` in arbitrary locations is difficult to debug since it depends on setting global state. It also makes testing difficult because if you want to inject the environment into a test as a dependency, this is harder because you have to always `evergreen.SetEnvironment` to ensure that all calls to `evergreen.GetEnvironment()` shares one global state (and later remember to un-set it or else it could affect future tests). It's usually preferable to call `evergreen.GetEnvironment()` once when the job starts running and pass that around.